handlers/handlers.go

```diff
@@ -48,3 +49,61 @@ func DoCheckCookie(h http.Handler, key Key) http.Handler {
 		h.ServeHTTP(w, req)
 	})
 }
+
+/* My head hurts...
+ControllerHandler is called from an app which then creates a function of type ControllerHandlerFunc which returns a http.Handler (and error)
+The former is then passed to the ControllerHandlerMiddleware function which checks all fields are present and accounted for
+then calls the ControllerHandlerFunc implementation which sets the headers as required to finally return a http.Handler back to the original app that called ControllerHandler
+*/
+//TODO all the error handling
+
+// TODO move type to top of file
+type ControllerHandlerFunc func(ctx context.Context, w http.ResponseWriter, r *http.Request, lang, collectionID, accessToken string) (http.Handler, error)
+
+/* Should ControllerHandlerFunc type be implemented here too or are the individual apps providing the
+'handlerFunc ControllerHandlerFunc' as seen on line 79? As the headers will have to be set somewhere. If here then line 68 shows how this might roughly be done
+*/
+
+//WIP - as above not 100% if this should be done here or in app but this is roughly what it could look like
```

Comment:
As you say, the actual handler implementation would live in the app, but the logic here whereby the closure is being returned is what we will want in the middleware (see comment above).